examples/model_selection/plot_precision_recall.py

```diff
@@ -152,23 +238,103 @@
 lines.append(l)
-labels.append('micro-average Precision-recall curve (area = {0:0.2f})'
+labels.append('micro-average Precision-recall (area = {0:0.2f})'
               ''.format(average_precision["micro"]))
+
 for i, color in zip(range(n_classes), colors):
-    l, = plt.plot(recall[i], precision[i], color=color, lw=lw)
+    l, = plt.plot(recall[i], precision[i], color=color, lw=2)
     lines.append(l)
-    labels.append('Precision-recall curve of class {0} (area = {1:0.2f})'
+    labels.append('Precision-recall for class {0} (area = {1:0.2f})'
                   ''.format(i, average_precision[i]))
 
 fig = plt.gcf()
-fig.set_size_inches(7, 7)
 fig.subplots_adjust(bottom=0.25)
 plt.xlim([0.0, 1.0])
 plt.ylim([0.0, 1.05])
 plt.xlabel('Recall')
 plt.ylabel('Precision')
 plt.title('Extension of Precision-Recall curve to multi-class')
-plt.figlegend(lines, labels, loc='lower center')
+plt.legend(lines, labels, loc=(0, -.38), prop=dict(size=14))
+
+
+###############################################################################
+# Eleven-point average precision
+# ------------------------------
+#
+# In *interpolated* average precision, a set of desired recall values is
+# specified and for each desired value, we average the best precision
+# scores possible with a recall value at least equal to the target value.
+# The most common choice is 'eleven point' interpolated precision, where
+# the desired recall values are [0, 0.1, 0.2, ..., 1.0]. This is the
+# metric referenced in `The PASCAL Visual Object Classes (VOC) Challenge
+# <http://citeseerx.ist.psu.edu
+# /viewdoc/download?doi=10.1.1.157.5766&rep=rep1&type=pdf>`_ (top of page
+# 11, formula 1). In the example below, the eleven precision values are
+# indicated with an arrow to pointing to the best precision possible
+# while meeting or exceeding the desired recall. Note that it's possible
+# that the same operating point might correspond to multiple desired
+# recall values.
+
+from operator import itemgetter
+
+
+def pick_eleven_points(recall_, precision_):
+    """Choose the eleven operating points that correspond
+    to the best precision for any ``recall >= r`` for r in
+    [0, 0.1, 0.2, ..., 1.0]
+    """
+    operating_points = list()
+    for target_recall in np.arange(0, 1.1, 0.1):
+        operating_points_to_consider = [pair
+                                        for pair in zip(recall_, precision_)
+                                        if pair[0] >= target_recall]
+        operating_points.append(max(operating_points_to_consider,
+                                    key=itemgetter(1)))
+    return operating_points
+
+# Work on the 2nd class of iris
+iris_cls = 2
+
+eleven_points = pick_eleven_points(recall[iris_cls], precision[iris_cls])
+interpolated_average_precision = np.mean([e[1] for e in eleven_points])
+
+
+print("Target recall    Selected recall   Precision")
+for i in range(11):
+    print("  >= {}           {:3.3f}             {:3.3f}".format(i / 10,
```

Comment:
Nitpick: if we're using format strings, maybe we should, you know, actually use them for the paddding?